utils/cups-browsed.c

```diff
@@ -711,9 +711,19 @@ long int findLogFileSize()
 }
 
 void copyToFile(FILE **fp1, FILE **fp2){
-  char ch;
-  while((ch = getc(*fp1)) != EOF)
-    putc(ch, *fp2);
+  int buffer_size = 2048;
+  char *buf = (char*) malloc(sizeof(char)*buffer_size);
+  if(!buf){
+    fprintf(stderr,"Error creating buffer for debug logging\n");
+    return;
+  }
+  fseek(*fp1, 0, SEEK_SET);
+  size_t r = fread(buf, sizeof(char), sizeof(buffer_size)-1, *fp1);
+  while(r == sizeof(buffer_size)-1){
+    fwrite(buf, sizeof(char), sizeof(buffer_size)-1, *fp2);
+    r = fread(buf, sizeof(char), sizeof(buffer_size)-1, *fp1);
+  }
+  fwrite(buf, sizeof(char), r, *fp2);
```

Comment:
You could simply do
```
do {
  r = fread(..., ..., buffer_size, ...);
  fwrite(..., ..., r, ...);
} while (r == buffer_size);
```
First, you can use the full buffer, no terminating zero is needed, we do not have strings here, so no `-1` on the buffer size, second, you have to give buffer_size itself, not sizeof(buffer_size), as the latter gives you the size of the interger variable which holds the buffer size, which is much smaller than the buffer and so you end up using a much smaller buffer than you originally wanted. My version also reduces the number of `fread` and `fwrite` lines.

Reply:
Yes sure, I actually intended to write sizeof(buf).